streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -683,20 +683,16 @@ trait ZStream[-R, +E, +A] extends Serializable { self =>
             _ <- self.foreachManaged { a =>
                   for {
                     latch <- Promise.make[Nothing, Unit]
+                    _     <- permits.acquire
                     innerStream = Stream
-                      .bracket(permits.acquire)(_ => permits.release)
-                      .flatMap(_ => ZStream.fromEffect(latch.succeed(())).flatMap(_ => f(a)))
+                      .bracket(latch.succeed(()))(_ => permits.release)
+                      .flatMap(_ => f(a))
                       .foreach(b => out.offer(Take.Value(b)).unit)
                       .foldCauseM(
-                        cause => out.offer(Take.Fail(cause)) *> innerFailure.fail(cause),
+                        cause => out.offer(Take.Fail(cause)) *> innerFailure.fail(cause).unit,
                         _ => ZIO.unit
                       )
                     _ <- (innerStream race interruptInners.await).fork
-                    // Make sure that the current inner stream has actually succeeded in acquiring
```

Comment:
Could we keep the comment and update it instead of dropping it entirely? :-)

Reply:
Ah yes, I forgot to put it back. Sorry.